doc/DHCPv4_Gateway/DHCPv4_gateway.md

```diff
@@ -0,0 +1,127 @@
+# DHCPv4 - Specify Gateway explicitly #
+
+## Table of Content 
+1. [Scope](#Scope)
+2. [Definitions](#Definitions/Abbreviations)
+3. [Overview](#Overview)
+4. [Requirements](#Requirements)
+5. [High Level Design](#High-Level-Design)
+6. [Test](#Testing-Considerations)
+
+### Revision  
+
+### Scope  
+
+This document describes the High Level Design of 'Subnets' of vlan and use of same in dhcpv4.
+This HLD document represent the changes in PR: https://github.com/sonic-net/sonic-buildimage/pull/15969.
+
+### Definitions/Abbreviations 
+
+
+
+### Overview 
+
+'Subnets' specify the primary subnet of a vlan (IPv4;IPv6). When a dhcpv4 packet is forwarded via dhcprelayagent, it embeds its own gateway address as 'giaddr' for return communication.
+In case of multiple interfaces, dhcprelayagent misconfigures this giaddr to any of the interfaces. 
+This change fixes this behaviour by explicitly specifying the primary interface/gateway.
+This primary subnet is stored in db as 'subnets' of VLAN.
+
+### Requirements
+
+1. Support a new member 'Subnets' of VLAN in config_db.
```

Comment:
Propose to modify INTERFACE_TABLE for specifying secondary IPs as this could extend the implementation for future use.

Reply:
updated the HLD and PRs according to this suggestion.